src/Service/AuditLog/Loggers/AuditLoggerFactory.cs

```diff
@@ -0,0 +1,27 @@
+using Passwordless.Service.Features;
+
+namespace Passwordless.Service.AuditLog.Loggers;
+
+public interface IAuditLoggerFactory
+{
+    Task<IAuditLogger> Create();
+}
+
+public class AuditLoggerFactory : IAuditLoggerFactory
+{
+    private readonly IFeatureContextProvider _provider;
+    private readonly INullAuditLogger _nullAuditLogger;
+    private readonly IAuditLogger _auditLogger;
+
+    public AuditLoggerFactory(IFeatureContextProvider provider, INullAuditLogger nullAuditLogger, IAuditLogger auditLogger)
+    {
+        _provider = provider;
+        _nullAuditLogger = nullAuditLogger;
+        _auditLogger = auditLogger;
+    }
+
+    public async Task<IAuditLogger> Create() =>

```

Comment:
We can likely move this into a scoped or transient injection where IAuditLogger would be resolved to either NullAuditLogger or AuditLogger

Reply:
This is handled with the AuditLoggerProvider. It can't exist inside DI because the feature context requires async/await.